.github/actions/issues-nudge.js

```diff
@@ -0,0 +1,72 @@
+const { Octokit } = require("octokit");
+
+const token = process.env.GITHUB_TOKEN;
+const [owner, repo] = process.env.GITHUB_REPOSITORY.split("/");
+
+const octokit = new Octokit({ auth: token });
+
+async function check_issue(issue) {
+  const events = await octokit.paginate(octokit.rest.issues.listEvents, {
+    owner: owner,
+    repo: repo,
+    issue_number: issue.number,
+    per_page: 100,
+  });
+  const added_to_project = events.filter(e => e.event == "added_to_project");
+  const removed_from_project = events.filter(e => e.event == "removed_from_project");
+  const projects = added_to_project.length - removed_from_project.length;
+
+  if (projects > 0) {
+    console.log("It appears issue #" + issue.number + " already has at least one project.");
+  } else {
+    console.log("It appears issue #" + issue.number + " has no project...");
+
+    const { data: issue_details } = await octokit.rest.issues.get({
+      owner: owner,
+      repo: repo,
+      issue_number: issue.number,
+    });
```

Comment:
Does the JS API not allow to specify *what* we want as response, GraphQL-style (or whatever their API format is called)? I.e. saying "and just give us assignees and closed_by"?

Reply:
Ah I guess that needs https://github.com/octokit/graphql.js/

Reply:
Yep, but what I find even more interesting is that GraphQL exposes the projects of an issue :astonished: :running_man:

Reply:
Let's say "it depends":
```graphql
query {
  search(query:"repo:root-project/root is:issue is:closed no:project", type:ISSUE, first:100) {
    nodes {
      ... on Issue {
        number
        assignees(first:10) {
          nodes {
            login
          }
        }
      }
    }
  }
}
```
does exactly what you think it does (although I have to say `... on Issue` is not very intuitive). The only problem is that they're not exposing the `closed_by` :sob: We could go with only the assignees, but I guess the better approach is to use GraphQL to query all issues without a project (that one even doesn't care if it's assigned to a card!) and then go with the existing approach to query single issues via the REST API. Still way faster than crawling through all thousands of issues and PRs plus their events

Reply:
> The only problem is that they're not exposing the `closed_by`

You see that also with the [GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer)?

Reply:
Looks like you want `ClosedEvent` (part of `IssueTimelineItem`) which has an `actor` field.

Reply:
Yeah, but then again we have to iterate through all events :disappointed:

Reply:
Beh. OK, too bad.

> I guess the better approach is to use GraphQL to query all issues without a project (that one even doesn't care if it's assigned to a card!) and then go with the existing approach to query single issues via the REST API.

seems to be the winner then.